app/lib/actions/pulp3/repository/save_version.rb

```diff
@@ -34,7 +34,12 @@ def run
               repo.update(:version_href => version_href)
             end
           else
-            output[:contents_changed] = false
+            # get publication and check if repo-version and publication match. Otherwise, contents_changed: false
+            if fetch_current_published_version_href(repo) != repo.version_href
```

Comment:
Since not all repository types have publications, this is causing `contents_changed` to be true for all syncs of those types. One example of a type without publications is the container/docker type.

I'd recommend checking if `repo.repository_type.pulp3_skip_publication` to fix the issue.

Reply:
Well spotted! I did not consider there are types without publications.

Reply:
Thanks, added a condition.